### ISSUE TITLE
V-IOU tracker improvements implementation ?

### ISSUE THREAD

**Issue by tdurand** [bochinski/iou-tracker#7]
Hello, 

Is the v-iou implementation available somewhere ? _(based on paper Extending IOU Based Multi-Object Tracking by Visual Information)_

I'm doing something close to V-IOU idea in a node.js implementation: https://github.com/tdurand/node-moving-things-tracker but in a much less "academic" way.

There is not much details about the "prediction" method you use for the visual tracking.. On my project I'm using velocity / acceleration vectors but do not change the bounding box width / height

From you paper, I'm gonna try this idea of searching _"back-wards through the last ttl frames for each new track"_ , this is something I'm not doing currently

Thanks !

Thibault

**Comment by bochinski** [comment]
Hi,

I had not time to cleanup and publish the code yet. I am not sure when this will happen, but it is definitely on my todo list.

For the prediction I am using a visual single object tracker (e.g. KCF is fast and reliable, at least for a couple of frames). Maybe a kalman filter is more suitable for your javascript application?

**Comment by tdurand** [comment]
Hi, thanks for your comment ! I'll look into KCF and Kalman filter... For now our implementation is giving good enough results but we didn't benchmark it really well. We have a very basic prediction method.

Feel free to close this if you want.. And I'm glad you will eventually publish the code at some point.

Best

**Comment by bochinski** [comment]
I will leave this open for reference since some people already asked about this by mail.

**Comment by jamessmith90** [comment]
@bochinski

Kalman is good but ...it is a cpu killer.
Using width,height and nearness with t-1 frames with time < 6 seconds will increase accuracy.

**Comment by lynnw123** [comment]
Congrs V-IOU is the top3 in the tracking challenge on MOT19 from yesterday's CVPR meeting, look forward to see its implementation!

**Comment by bochinski** [comment]
@jamessmith90 
I agree, but nothing is for free. Visual single object tracking/optical flow works much better (especially when the target movement is non-linear) but is much more cpu intensive. It really depends on what you want to achieve and what you can settle with.

@lynnw123 
thank you very much, unfortunately I was not able to attend this years CVPR on short notice but I assume it was a great workshop.
Anyway, this is a good motivation for me to publish the code sooner than later.

**Comment by jamessmith90** [comment]
@bochinski 
Looking forward to your v-iou implementation.

**Comment by dexception** [comment]
@bochinski 
Congrats for your position in MOT19.

Do you have plans to release the updated version ?

**Comment by bochinski** [comment]
@dexception thanks!

I already started working on the code but wasnt able to finish it yet as something else came in between. 
Hopefully there are no more problems/delays so you can expect the release of the code very soon.

**Comment by dexception** [comment]
@bochinski 
Can you share a timeline with us ?

**Comment by suman-19** [comment]
@bochinski 
Kindly provide some info.

**Comment by bochinski** [comment]
@dexception @suman-19 Sorry about the delay, I know that a few people are waiting on this. My original time line was to do this months ago but every time I want to start on this something else comes in between. I will try to do this sometime in the next weeks and I sincerely hope that I can keep my promise this time.

**Comment by dexception** [comment]
Thanks for the update.

**Comment by dexception** [comment]
@bochinski Any progress ?

**Comment by jamessmith90** [comment]
@bochinski Any update ?

**Comment by nicolaerosia** [comment]
I'm also very interested in this

**Comment by ashuezy** [comment]
@bochinski 7 months and still waiting for a release.. :-)

**Comment by uday60** [comment]
Waiting..

**Comment by penolove** [comment]
waiting .. :)

**Comment by sachingoyal0104** [comment]
waiting..

**Comment by bochinski** [comment]
I finally managed to push the v-iou code. I am sorry for the delay, it took way longer than anticipated. One reason for this is that i decided to completely rewrite my original code to get a clean version that is similar to the iou tracker code. Let me know if you encounter any issues, the results should be the same as in the paper and challenge submissions.

**Comment by nicolaerosia** [comment]
thank you!

**Comment by akshu281** [comment]
Hello All,

Any idea how to run this tracker V-IOU for a custom video with real-time detection ? 

Thank You !
